### PR TITLE
Implement `kelrond.sh` in Python

### PR DESCRIPTION
**Pull request by tothtamas28**
* Implement commands `kelrond {run,kast}` in Python
* Remove `kelrond.sh`

### CODE REVIEW

File: kmultiversx/src/kmultiversx/kelrond.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    program_text = program.read_text()
+    with NamedTemporaryFile() as f:
+        tmp_file = Path(f.name)
+        tmp_file.write_text(preprocess(program_text))
+        yield tmp_file
+
+
+def _exit_with_output(cp: CompletedProcess) -> None:
+    status = cp.returncode
+    out = cp.stdout if status else cp.stderr
```
Comment (bbyalcinkaya) [inline kmultiversx/src/kmultiversx/kelrond.py:73]: Shouldn't this be the opposite? It prints `cp.stderr` when the execution is successful. Same applies to the next line, too.

File: kmultiversx/src/kmultiversx/kelrond.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    raise AssertionError()
+
+
+def _exec_run(*, program: Path, backend: Backend, main_file: MainFile) -> None:
+    definition_dir = kdist.get(f'mx-semantics.{backend.value}-{main_file.value}')
+
+    with _preprocessed(program) as input_file:
+        proc_res = _krun(definition_dir=definition_dir, input_file=input_file, check=False)
+
+    _exit_with_output(proc_res)
```
Comment (bbyalcinkaya) [inline kmultiversx/src/kmultiversx/kelrond.py:50]: I think, it is better if we get both stderr and stdout of `krun` in the output. Mandos scenarios typically starts with `setExitCode 1` and ends with `setExitCode 0`. When a test fails, the return code is 1, but `stderr` might be empty. In such cases, we would need the final configuration to understand the cause of the failure.

File: kmultiversx/src/kmultiversx/kelrond.py (hunk 1 above)
Comment (tothtamas28) [reply kmultiversx/src/kmultiversx/kelrond.py:73]: You're right, good catch!

File: kmultiversx/src/kmultiversx/kelrond.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    kast_parser = command_parser.add_parser('kast', help='parse a Mandos scenario and output it in a supported format')
+    _add_common_arguments(kast_parser)
+    kast_parser.add_argument('--output', metavar='FORMAT', type=KAstOutput, help='format to output the term in')
+
+    return parser
+
+
+def _add_common_arguments(parser: ArgumentParser) -> None:
+    parser.add_argument('program', metavar='PROGRAM', type=file_path, help='path to WebAssembly program')
```
Comment (bbyalcinkaya) [inline kmultiversx/src/kmultiversx/kelrond.py:94]:
```suggestion
    parser.add_argument('program', metavar='PROGRAM', type=file_path, help='path to Mandos scenario')
```

File: kmultiversx/src/kmultiversx/kelrond.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+def _exit_with_output(cp: CompletedProcess) -> None:
+    status = cp.returncode
+    out = cp.stdout if status else cp.stderr
+    file = sys.stdout if status else sys.stderr
+    print(out, end='', file=file, flush=True)
+    sys.exit(status)
+
+
+def _argument_parser() -> ArgumentParser:
+    parser = ArgumentParser(prog='kwasm')
```
Comment (bbyalcinkaya) [inline kmultiversx/src/kmultiversx/kelrond.py:80]:
```suggestion
    parser = ArgumentParser(prog='kelrond')
```

**Review by bbyalcinkaya — CHANGES_REQUESTED** [review]

File: kmultiversx/src/kmultiversx/kelrond.py (hunk 2 above)
Comment (tothtamas28) [reply kmultiversx/src/kmultiversx/kelrond.py:50]: Done.

**Review by bbyalcinkaya — APPROVED** [review]

**Pull request merged**